api/datasets.go

```diff
@@ -34,7 +35,7 @@ type DatasetHandlers struct {
 func NewDatasetHandlers(inst *lib.Instance, readOnly bool) *DatasetHandlers {
 	dsm := lib.NewDatasetMethods(inst)
 	rm := lib.NewRemoteMethods(inst)
-	h := DatasetHandlers{*dsm, rm, inst.Node(), inst.Node().Repo, readOnly}
+	h := DatasetHandlers{*dsm, inst, rm, inst.Node(), inst.Node().Repo, readOnly}
```

Comment:
While you're here, could you also remove the node and repo fields? I was going to say you could just get them from the instance, but it turns out they're not used at all.

Reply:
wooo def, that's going to feel v good to remove.